### ISSUE TITLE
Sensor override broken after 3e35c761b7 (and maybe before that too)

### ISSUE THREAD

**Issue by vsytch** [openbmc/bmcweb#253]
**Describe the bug**
[Changes](https://github.com/openbmc/bmcweb/commit/3e35c761b7793696267870550bd6f0b0035f4110) to the format of Sensor's `"MemberId"` field broke override logic due to it relying on the value of `"MemberId"` being a subset of the sensors name. 

[Here](https://github.com/openbmc/bmcweb/blob/master/redfish-core/lib/sensors.hpp#L2628) we will read a digit, but [further down](https://github.com/openbmc/bmcweb/blob/master/redfish-core/lib/sensors.hpp#L2642) we'll parse the digit as if its still the sensor name. How are we supposed to now match the new `"MemberId"` with the actual sensor?

**To Reproduce**
1. Run `curl --data '{"Fans":[{"MemberId": "0", "Reading":80}]}' -H "content-type: application/json" -X PATCH localhost/redfish/v1/Chassis/tray/Therma#/Fans/`
2. See error `[INFO "sensors.hpp":2684] Unable to find memberId 0`

**Is this a regression**
The part described above was broken by https://github.com/openbmc/bmcweb/commit/3e35c761b7793696267870550bd6f0b0035f4110, however reverting that commit only helps a bit. bmcweb is able to find the sensor, but fails dbus queries to fansensor.
Running:
```
curl --data '{"Fans":[{"MemberId": "fanpwm_fan0_pwm", "Reading":80}]}' -H "content-type: application/json" -X PATCH localhost/redfish/v1/Chassis/chassis/Thermal#/Fans/
```
Results in:
```
[DEBUG "sensors.hpp":385] sensorNames list count: 1
[DEBUG "sensors.hpp":388] Sensor to find: /xyz/openbmc_project/sensors/fan_pwm/fan0_pwm
[DEBUG "sensors.hpp":401] Adding connection: xyz.openbmc_project.FanSensor
[DEBUG "sensors.hpp":408] Found 1 connections
[INFO "sensors.hpp":2725] Unable to find sensor object/xyz/openbmc_project/sensors/fan_pwm/fan0_pwm
[CRITICAL "error_messages.cpp":282] Internal Error /usr/src/debug/bmcweb/1.0+git999-r0/redfish-core/lib/sensors.hpp(2727:44) `redfish::setSensorsOverride(const std::shared_ptr<SensorsAsyncResp>&, std::unordered_map<std::__cxx11::basic_string<char>, std::vector<nlohmann::json_abi_v3_11_2::basic_json<>, std::allocator<nlohmann::json_abi_v3_11_2::basic_json<> > > >&)::<lambda(const std::shared_ptr<std::set<std::__cxx11::basic_string<char> > >&)>::<lambda(const std::set<std::__cxx11::basic_string<char> >&, const std::set<std::pair<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> > >&)>`:
[DEBUG "sensors.hpp":410] getObjectsWithConnection resp_handler exit
```
While fansensor is all well:
```
busctl introspect xyz.openbmc_project.FanSensor /xyz/openbmc_project/sensors/fan_pwm/fan0_pwm
NAME                                        TYPE      SIGNATURE RESULT/VALUE                             FLAGS
org.freedesktop.DBus.Introspectable         interface -         -                                        -
.Introspect                                 method    -         s                                        -
org.freedesktop.DBus.Peer                   interface -         -                                        -
.GetMachineId                               method    -         s                                        -
.Ping                                       method    -         -                                        -
org.freedesktop.DBus.Properties             interface -         -                                        -
.Get                                        method    ss        v                                        -
.GetAll                                     method    s         a{sv}                                    -
.Set                                        method    ssv       -                                        -
.PropertiesChanged                          signal    sa{sv}as  -                                        -
xyz.openbmc_project.Association.Definitions interface -         -                                        -
.Associations                               property  a(sss)    1 "chassis" "all_sensors" "/xyz/openbmc… emits-change
xyz.openbmc_project.Sensor.Value            interface -         -                                        -
.MaxValue                                   property  d         100                                      emits-change
.MinValue                                   property  d         0                                        emits-change
.Unit                                       property  s         "xyz.openbmc_project.Sensor.Value.Unit.… emits-change
.Value                                      property  d         29.8039                                  emits-change writable
```

**Comment by edtanous** [comment]
ACK, yeah, that looks like a really unfortunate side effect that I certainly didn't expect when merging that code.  In terms of priority, I think we did get that right.  A failing service validator effects _EVERY_ platform, so if the two requirements are mutually exclusive, I'd much rather have a passing service validator than have sensor override (which I don't think is actually enabled on anything but downstream platforms).

With that said, lets see if we can get it working for both.  I think to fix that, it's going to require us to go back to building the entire sensor tree, to determine the index of the sensor in question, then firing it to that path.  Let me know if you need help rearranging the code.

**Comment by edtanous** [comment]
That's probably reasonable.  Please make sure the validator passes on whatever change we make.

**Comment by vsytch** [comment]
As a short term fix, reverting 3e35c761b7793696267870550bd6f0b0035f4110 and c71d61258bd3cd8573166011b450a1eecce2c572 works.
```
curl localhost/redfish/v1/Chassis/Tray/Thermal#/Fans
...
    {
      "@odata.id": "/redfish/v1/Chassis/Tray/Thermal#/Fans/0",
      "@odata.type": "#Thermal.v1_3_0.Fan",
      "MaxReadingRange": 100,
      "MemberId": "fan0_pwm",
      "MinReadingRange": 0,
      "Name": "fan0 pwm",
      "Reading": 29,
      "ReadingUnits": "Percent",
      "Status": {
        "Health": "OK",
        "State": "Enabled"
      }
    },
...
```
```
curl --data '{"Fans":[{"MemberId": "fan0_pwm", "Reading": 50}]}' -X PATCH localhost/redfish/v1/Chassis/Tray/Thermal#/Fans/
{
  "@odata.id": "/redfish/v1/Chassis/Tray/Thermal",
  "@odata.type": "#Thermal.v1_4_0.Thermal",
  "Fans": [],
  "Id": "Thermal",
  "Name": "Thermal",
  "Temperatures": []
}
```
```
curl localhost/redfish/v1/Chassis/Tray/Thermal#/Fans
...
    {
      "@odata.id": "/redfish/v1/Chassis/Tray/Thermal#/Fans/0",
      "@odata.type": "#Thermal.v1_3_0.Fan",
      "MaxReadingRange": 100,
      "MemberId": "fan0_pwm",
      "MinReadingRange": 0,
      "Name": "fan0 pwm",
      "Reading": 50,
      "ReadingUnits": "Percent",
      "Status": {
        "Health": "OK",
        "State": "Enabled"
      }
    },
...
```
Based on the second commit, shouldn't have it "fixed" sensor overrides? But instead its causing the dbus error I mentioned previously.

**Comment by edtanous** [comment]
This was resolved with https://gerrit.openbmc.org/c/openbmc/bmcweb/+/64528
